### PR TITLE
improvements to ucx build from conda

### PR DESCRIPTION
**Pull request by nirandaperera**

### CODE REVIEW

**Review by Sandeepa1995 — DISMISSED** [review]

**Comment by nirandaperera** [comment]
@supunkamburugamuve any idea why the MacOS build is failing?

**Comment by supunkamburugamuve** [comment]
@nirandaperera this is due to some strange thing in Github actions macos VM. We need to just do the rm '/usr/local/bin/2to3' in our script. I encountered the same with the windows-build branch and fixed it like that there.

**Comment by supunkamburugamuve** [comment]
@nirandaperera lets first merge the windows pull request and then this one should compile

File: cpp/CMake/Modules/ConfigureArrow.cmake [hunk 1]
```diff
@@ -55,14 +55,13 @@ set(ARROW_CMAKE_ARGS " -DARROW_WITH_LZ4=OFF"
 
 if (PYCYLON_BUILD)
     find_package(Python3 COMPONENTS Interpreter REQUIRED)
-    message("Python Executable Path ${Python3_EXECUTABLE}")
+    message(STATUS "Python Executable Path ${Python3_EXECUTABLE}")
     if(WIN32)
```
Comment (supunkamburugamuve) [inline cpp/CMake/Modules/ConfigureArrow.cmake:59]: is this check not needed now?

File: cpp/CMake/Modules/ConfigureGlog.cmake [hunk 2]
```diff
@@ -60,7 +58,7 @@ set(GLOG_INCLUDE_DIR "${GLOG_ROOT}/install/include")
 set(GLOG_FOUND TRUE)
 
 IF(WIN32)
-        set(GLOG_LIBRARIES ${GLOG_INSTALL}/lib/glogd.lib)       
+        set(GLOG_LIBRARIES ${GLOG_INSTALL}/lib/glogd.lib)
```
Comment (supunkamburugamuve) [inline cpp/CMake/Modules/ConfigureGlog.cmake:61]: Lets use te glog.lib and not the debug one. These are from the previous pull request. But lets fix those here

File: cpp/CMake/Templates/Glog.CMakeLists.txt.cmake [hunk 3]
```diff
@@ -22,4 +22,4 @@ ExternalProject_Add(glog
         SOURCE_DIR        "${GLOG_ROOT}/glog"
         BINARY_DIR        "${GLOG_ROOT}/build"
         INSTALL_DIR       "${GLOG_ROOT}/install"
-        CMAKE_ARGS        -DCMAKE_INSTALL_PREFIX=${GLOG_ROOT}/install -DWITH_GFLAGS=OFF -DBUILD_SHARED_LIBS=OFF)
\ No newline at end of file
+        CMAKE_ARGS        -DCMAKE_INSTALL_PREFIX=${GLOG_ROOT}/install -DWITH_GFLAGS=OFF -DWITH_UNWIND=OFF -DBUILD_SHARED_LIBS=OFF)
```
Comment (supunkamburugamuve) [inline cpp/CMake/Templates/Glog.CMakeLists.txt.cmake:25]: Lets set the build type to release here

**Review by supunkamburugamuve — APPROVED** [review]

**Pull request merged**